### PR TITLE
Update interceptor to allow authorization header to be overridden

### PR DESCRIPTION
**Pull request by ruipenso**
Refs #57.
Refs #74.

### CODE REVIEW

File: src/interceptors/oauth-interceptor.js [hunk 1]
```diff
@@ -8,9 +8,10 @@
 function oauthInterceptor($q, $rootScope, OAuthToken) {
   return {
     request: function(config) {
+      config.headers = config.headers || {};
+
       // Inject `Authorization` header.
-      if (OAuthToken.getAuthorizationHeader()) {
-        config.headers = config.headers || {};
+      if (!config.headers.hasOwnProperty('Authorization') && OAuthToken.getAuthorizationHeader()) {
```
Comment (pgom) [inline src/interceptors/oauth-interceptor.js]: Lodash is used, so why not `!_.has(config, 'Authorization')`?

Comment (ruipenso) [inline src/interceptors/oauth-interceptor.js]: We don't use lodash.

Comment (pgom) [inline src/interceptors/oauth-interceptor.js]: I saw `lodash` as dependency on `package.json` and jumped too fast on this :disappointed_relieved:

**Comment by VyTacka** [comment]
After this fix if you'll refresh token and after that you'll try to send old request which was waiting in a queue for refreshing token, it'll send an old access token.

**Comment by ruipenso** [comment]
@VyTacka Good catch. We need to fix this and bump the major version for this change.

For now, I suggest removing the `Authorization` header before replying the request.

**Pull request merged**